inspirehep/dojson/utils.py

```diff
@@ -131,15 +131,27 @@ def remove_duplicates_from_list_of_dicts(ld):
     """Remove duplicates from a list of dictionaries preserving the order.
 
     We can't use the generic list helper because a dictionary isn't
-    hashable. Taken from http://stackoverflow.com/a/9427216/374865."""
+    hashable. Adapted from http://stackoverflow.com/a/9427216/374865."""
+
+    def _freeze(o):
+        """Recursively freezes a dict in an hashable object.
+
+        Adapted from http://stackoverflow.com/a/21614155/374865."""
+        if isinstance(o, dict):
+            return frozenset({k: _freeze(v) for k, v in o.items()}.items())
+        elif isinstance(o, (list, tuple)):
+            return tuple([_freeze(v) for v in o])
+        else:
+            return o
```

Comment:
BTW, I think we should move this as `before_record_insert/update` receiver. This is because we'll need this also when we cut the rope from legacy.